### ISSUE TITLE
Volume for PostgreSQL is always created even though project uses MariaDB for the database

### ISSUE THREAD

**Issue by alexkuc** [ddev/ddev#5880]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
[COPY-PASTE HERE THE OUTPUT OF `ddev debug test`]
```
</details>         

### Expected Behavior

Only one database volume to be created with the name to match what is specified in the project's config

### Actual Behavior

I am seeing two database volumes - one for PostgreSQL and one for MariaDB - even though the project specifies:

```yaml
database:
    type: mariadb
    version: '10.4'
```

```shell
docker volume ls

DRIVER    VOLUME NAME
local     ddev-<project>-snapshots
local     ddev-global-cache
local     ddev-ssh-agent_dot_ssh
local     ddev-ssh-agent_socket_dir
local     <project>-mariadb
local     <project>-postgres
```

### Steps To Reproduce

Following [these steps](https://ddev.readthedocs.io/en/stable/users/quickstart/#wordpress) to create standard WordPress project

### Anything else?

I have not included output from `ddev debug test` because I am not certain if it is expected to be like that or if it's an actual bug. If this output is still required, I would be more than happy to include it as well by editing this issue.

**Comment by rfay** [comment]
It creates an empty volume is all. It doesn't do anything with it. 

The other thing you'll note is that the volume used for both mariadb and mysql is called "<project>-mariadb"

If you want to take a look and make the code better and you don't break anything, your contribution will be welcome. 

It's easier than your repro to set this up:

```
mkdir junk && cd junk
ddev config --auto
ddev start
docker volume ls | grep junk
```

However, if you mount the postgres volume you'll find it's empty and harmless. 

I'd say this is a shortcoming, a shortcut I took when implementing postgres. If somebody wants to make it better their contribution will be welcome.

**Comment by alexkuc** [comment]
> It creates an empty volume is all. It doesn't do anything with it.

> However, if you mount the postgres volume you'll find it's empty and harmless.

To be honest, I am not a perfectionist so if it doesn't occupy space or actually do anything, fine by me. My concern was that these "redundant" volumes would continue to grow in size causing increased usage of disk space. Unfortunately, with docker, it is too easy to run out of space unintentionally.

**Comment by alexkuc** [comment]
> If you want to take a look and make the code better and you don't break anything, your contribution will be welcome.

As much as I'd like to, I have absolute zero knowledge of Go and my time schedule does not permit currently. Would it be okay to leave this issue open in case someone else wishes to try their luck? Or is it better to close it?

**Comment by rfay** [comment]
Yes, I was considering closing, but it would be reasonable to fix this. Nobody has ever noticed the creation of the volume before, but people have commented on the output during `ddev delete` that shows it being deleted.

**Comment by rfay** [comment]
However... there's no real reason to remove a zero-size volume. But whatever!

**Comment by rfay** [comment]
This is a worthy issue :)

**Comment by stasadev** [comment]
This is already fixed:

- #7937
